src/iterated_integrals.jl

```diff
@@ -0,0 +1,286 @@
+    m_seq::Array{Int}
+end
+
+mutable struct WikJGeneral_iip{rateNoiseElTypeNoUnits, WikJType} <: AbstractWikJGeneral
+    WikJ::WikJType
+    WikJ2::WikJType
+    WikJ3::WikJType
+    m_seq::Array{Int}
+    vec_ζ::Vector{eltype(rateNoiseElTypeNoUnits)}
+    vec_η::Vector{eltype(rateNoiseElTypeNoUnits)}
+    Gp1::Vector{eltype(rateNoiseElTypeNoUnits)}
+    Gp2::Vector{eltype(rateNoiseElTypeNoUnits)}
+    Aᵢ::Vector{eltype(rateNoiseElTypeNoUnits)}
+end
+
+function fill_WikJDiagonal_oop(ΔW)
+    WikJDiagonal_oop()
+end
+
+function fill_WikJDiagonal_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW
+    WikJDiagonal_iip{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJCommute_oop(ΔW)
+    WikJCommute_oop()
+end
+
+function fill_WikJCommute_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJCommute_iip{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJGeneral_oop(ΔW)
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    WikJGeneral_oop(m_seq)
+end
+
+function fill_WikJGeneral_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJ2 = false .* ΔW .* ΔW'
+    WikJ3 = false .* ΔW .* ΔW'
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    vec_ζ = false .* vec(ΔW)
+    vec_η = false .* vec(ΔW)
+    Gp1 = false .* Array{eltype(ΔW)}(undef, M)
+    Gp2 = false .* Array{eltype(ΔW)}(undef, M)
+    Aᵢ = false .* vec(ΔW)
+    WikJGeneral_iip{eltype(ΔW), typeof(WikJ)}(WikJ, WikJ2, WikJ3, m_seq, vec_ζ, vec_η, Gp1, Gp2, Aᵢ)
+end
+
+function get_iterated_I!(dW, Wik::WikJDiagonal_oop, C=1)
+    WikJ = 1//2 .* dW .* dW
+    WikJ
+end
+
+function get_iterated_I!(dW, Wik::WikJDiagonal_iip, C=1)
+    @unpack WikJ = Wik
+    if typeof(dW) <: Number
+        Wik.WikJ = 1//2 .* dW .^ 2
+    else
+        @.. WikJ = 1//2*dW^2
+    end
+    return nothing
+end
+
+function get_iterated_I!(dW, Wik::WikJCommute_oop, C=1)
+    WikJ = 1//2 .* vec(dW) .* vec(dW)'
+    WikJ
+end
+
+function get_iterated_I!(dW, Wik::WikJCommute_iip, C=1)
+    @unpack WikJ = Wik
+    mul!(WikJ,vec(dW),vec(dW)')
+    @.. WikJ *= 1//2
+    return nothing
+end
+
+function get_iterated_I!(dW, Wik::WikJGeneral_oop, C=1)
+    @unpack m_seq = Wik
+    m      = length(dW)
+    M      = m*(m-1)/2
+    sum_dW² = dW'*dW
+
+    WikJ = dW*dW'
+    Gp1 = randn(M)
+    α = sqrt(1 + sum_dW²/dt)
+    Gp2 = Gp1/(sqrt(2)*(1+α)*dt)
+
+    #operator (Iₘ² - Pₘ)Kₘᵀ
+    for i in 1:M
+        WikJ2[m_seq[i,1], m_seq[i,2]] = Gp2[i]
+        WikJ2[m_seq[i,2], m_seq[i,1]] = -Gp2[i]
+    end
+
+    #operator (Iₘ X W*Wᵀ)
+    WikJ2 = WikJ*WikJ2
+
+    #operator Kₘ(Iₘ² - Pₘ)
+    WikJ2 = WikJ2 - WikJ2'
+    for i in 1:M
+        Gp2[i] = WikJ2[m_seq[i,1], m_seq[i,2]]
+    end
+    Gp = Gp/sqrt(2) + Gp2
```

Comment:
Gp on the right hand side here is undefined? Also, where does this relation come from?

Reply:
![image](https://user-images.githubusercontent.com/26359940/62858818-0196c180-bd19-11e9-9e24-9a31d48ecc8d.png)

So what I'm actually doing is applying the Square root sigma infinity operator on `Gp`. So this can be broken down into `I_M` part and the remaining part.
So when I'm saying ` Gp1 = Gp1/sqrt(2) + Gp2` here `Gp1` is the initial `Gp1` as `I_M * Gp1` will still be `Gp` and `Gp2` is the rest of the operator applied to `Gp1`. Thus this expression will give the Sigma_Infinity Operator of `Gp1`.